heudiconv/convert.py

```diff
@@ -534,14 +653,27 @@ def save_converted_files(res, item_dicoms, bids_options, outtype, prefix, outnam
         #   series. To do that, the most straightforward way is to read the
         #   echo times for all bids_files and see if they are all the same or not.
 
-        # Check for varying echo times
-        echo_times = sorted(list(set(
-            b.get('EchoTime', nan)
-            for b in bids_metas
-            if b
-        )))
-
-        is_multiecho = len(echo_times) > 1
+        # Collect some metadata across all images
+        echo_times, channel_names, image_types = [], [], []
+        for metadata in bids_metas:
+            if not metadata:
+                continue
+            echo_times.append(metadata.get('EchoTime', None))
+            channel_names.append(metadata.get('CoilString', None))
+            image_types.append(metadata.get('ImageType', None))
+        echo_times = [v for v in echo_times if v]
+        echo_times = sorted(list(set(echo_times)))
+        channel_names = [v for v in channel_names if v]
+        channel_names = sorted(list(set(channel_names)))
+        image_types = [v for v in image_types if v]
+
+        is_multiecho = len(echo_times) > 1  # Check for varying echo times
+        is_uncombined = len(channel_names) > 1  # Check for uncombined data
+
+        # Determine if data are complex (magnitude + phase)
+        magnitude_found = any(['M' in it for it in image_types])
+        phase_found = any(['P' in it for it in image_types])
```

Comment:
minor, also no need for an explicit `[]` list -- `any('P' in it for it in image_types)` would be just fine since you would just get an iterator comprehension now

Reply:
actually since we do not care about those `_found` variables and actually about retaining the list of lists for `image_types`, and at the end just getting `sets` of values:
-  start with sets, not lists and either `.add`  or `.update` (for image_types) - would remove need for all the sorted and explicit set.
- And then here it would just become a 
`is_multiecho = len(filter(bool, echo_times)) > 1` (or `is_multiecho = len(v for v in echo_times if v)) > 1` if current form)
`is_complex = 'M' in image_types and 'P' in image_types`

Reply:
It looks like it has to be `is_multiecho = len(set(filter(bool, echo_times))) > 1`, but it's now in there and looks much cleaner.